### PR TITLE
feat: build napi modules

### PR DESCRIPTION
**Pull request by Tuditi**
# Description of change

Wallet.rs is now packaged as N-API modules instead of electron and node modules. This gets rid of duplications and brings us one step closer to have context-aware native modules. Unfortunately, this still crashes the renderer process if reloaded twice.

## Links to any relevant issues

#1382

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Built index.node using `npm run prebuild` and replaced it in the existing Firefly

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/covector-version-or-publish.yml [hunk 1]
```diff
@@ -51,57 +51,31 @@ jobs:
...
-          node-version: ${{ matrix.node-version }}
+          node-version: 16.x
 
       - name: Install Rust toolchain
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
           override: true
       
-      - name: Install LLVM and Clang (Windows) # required for bindgen to work, see https://github.com/rust-lang/rust-bindgen/issues/1797
```
Comment (Thoralf-M) [inline .github/workflows/covector-version-or-publish.yml]: That's really not required anymore?

Comment (Tuditi) [reply .github/workflows/covector-version-or-publish.yml]: It is required. Only the different node versions can be omitted.

File: .github/workflows/covector-version-or-publish.yml [hunk 2]
```diff
@@ -61,11 +61,8 @@ jobs:
         # The GitHub hosted Windows 2022 image comes with Visual Studio 2022, but node-gyp
         # (which is used by neon-sys) sadly fails to recognize it. As a mitigation, we still run the
         # tests on Windows 2019, until we can figure out a way to fix the problem.
-        os: [ubuntu-latest, macos-latest, windows-2019]
-        node-version: ['14.x', '16.x', '18.x']
-        exclude:
-          - os: ubuntu-latest
-            node-version: '14.x'
+        os: [ubuntu-20.04, macos-11, windows-2019]
```
Comment (Thoralf-M) [inline .github/workflows/covector-version-or-publish.yml:64]: Why the change to the older versions for ubuntu and macos?

Comment (Tuditi) [reply .github/workflows/covector-version-or-publish.yml:64]: Because this make sure it's compatible with older devices as well as newer ones while still being properly supported by GHA.

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]
